src/frontend/src/pages/PIAForm/index.tsx

```diff
@@ -349,111 +348,107 @@ const PIAFormPage = () => {
   // Event Handlers
   //
 
+  /**
+   * Populate the PIA modal with data from the provided modal object.
+   * @param {object} modal - The modal object containing data to populate the modal for submit all status except PIAIntake.
+   */
   const populateModalFn = (modal: object) => {
     setPiaModalTitleText(Object(modal).title);
     setPiaModalParagraph(Object(modal).description);
     setPiaModalConfirmLabel(Object(modal).confirmLabel);
     setPiaModalCancelLabel(Object(modal).cancelLabel);
   };
 
+  /**
+   * Handle the display of modals based on the given modal type.
+   * @param {string} modalType - The type of modal to display.
+   * @param {string} conflictUser - Optional conflict user (used for 'conflict' modal).
+   */
   const handleShowModal = (modalType: string, conflictUser = '') => {
     switch (modalType) {
-      case 'cancel':
-        setPiaModalConfirmLabel(Messages.Modal.Cancel.ConfirmLabel.en);
-        setPiaModalCancelLabel(Messages.Modal.Cancel.CancelLabel.en);
-        setPiaModalTitleText(Messages.Modal.Cancel.TitleText.en);
-        setPiaModalParagraph(Messages.Modal.Cancel.ParagraphText.en);
-        setPiaModalButtonValue('cancel');
-        break;
-      case 'save':
-        setPiaModalConfirmLabel(Messages.Modal.Save.ConfirmLabel.en);
-        setPiaModalCancelLabel(Messages.Modal.Save.CancelLabel.en);
-        setPiaModalTitleText(Messages.Modal.Save.TitleText.en);
-        setPiaModalParagraph(Messages.Modal.Save.ParagraphText.en);
-        setPiaModalButtonValue('save');
-        break;
-      case 'edit':
+      case 'edit': {
         /* Using the state table, we can determine which modal to show based on the status of the PIA
            This will keep the modal text in one place and allow for easy updates in the future
            and will make the whole app consistent.
         */
-        PopulateModal(pia, PiaStatuses.EDIT_IN_PROGRESS, populateModalFn);
-        setPiaModalButtonValue('edit');
+        PopulateModal(
+          pia,
+          PiaStatuses.EDIT_IN_PROGRESS,
+          populateModalFn,
+          'submit',
```

Comment:
i would have this as as the current status which is the previous status. Having from == 'submit' is not scalable. Tomorrow there may be another button for e.g. 'next steps' and this may break.